src/TestFramework/PhpUnit/Config/Builder/MutationConfigBuilder.php

```diff
@@ -29,68 +29,70 @@ class MutationConfigBuilder extends ConfigBuilder
     private $projectDir;
 
     /**
-     * @var \DOMDocument
+     * @var XmlConfigurationHelper
      */
-    private $dom;
+    private $xmlConfigurationHelper;
 
     /**
-     * @var \DOMXPath
+     * @var string
      */
-    private $xPath;
+    private $originalXmlConfigContent;
 
     public function __construct(string $tempDirectory, string $originalXmlConfigContent, XmlConfigurationHelper $xmlConfigurationHelper, string $projectDir)
     {
         $this->tempDirectory = $tempDirectory;
         $this->projectDir = $projectDir;
 
-        $this->dom = new \DOMDocument();
-        $this->dom->preserveWhiteSpace = false;
-        $this->dom->formatOutput = true;
-        $this->dom->loadXML($originalXmlConfigContent);
-
-        $this->xPath = new \DOMXPath($this->dom);
-
-        $xmlConfigurationHelper->replaceWithAbsolutePaths($this->xPath);
-        $xmlConfigurationHelper->setStopOnFailure($this->xPath);
-        $xmlConfigurationHelper->deactivateColours($this->xPath);
-        $xmlConfigurationHelper->removeExistingLoggers($this->dom, $this->xPath);
-        $xmlConfigurationHelper->removeExistingPrinters($this->dom, $this->xPath);
+        $this->xmlConfigurationHelper = $xmlConfigurationHelper;
+        $this->originalXmlConfigContent = $originalXmlConfigContent;
     }
 
     public function build(MutantInterface $mutant): string
     {
+        $dom = new \DOMDocument();
```

Comment:
I have no idea how expensive the creation of these objects are, but maybe we can only do the steps that are different per mutant, every single time. While working from a base dom object

Reply:
Initially, I moved this object creation from the constructor by purpose, because there was another bug: https://travis-ci.org/infection/infection/jobs/372222312#L761

When this object was created in constructor, the same instance was used in `build()` method, and each time [it was modified](https://github.com/infection/infection/pull/323/files#diff-5780b7fc216dd1e72bb1cdde6558f339R125).

This led to side effect, when each new `phpunit.xml` file has previous one as an `bootstrap` file path (because \DOMDocument is not immutable).

So I decided to move it to `build()` method.

Now, with [this new commit](https://github.com/infection/infection/pull/323/commits/fab5839f74e19bc0c51229e56330b0cd919d1929), it's a little bit faster to clone already instantiated object rather than create it from the scratch. So I moved it back, but instead of the full reusing - it is being cloned.

Perfomance:

| Operation | Time |
|--|--|
|Instantiating `\DOMDocument`| Clone `\DOMDocument` |
| `8.0108642578125E-5ms` | from `9.0599060058594E-6ms` to `2.1934509277344E-5ms` |

So, cloning from 4 to 10 times faster.

Reply:
Pretty cool, how did you measure those times? just with `microtime()`, or do you use something else?

Reply:
Just microtime